api-test/hasura/actions/_handlers/createEpoch.test.ts

```diff
@@ -545,13 +548,182 @@ describe('createEpoch', () => {
     });
   });
   describe('monthly input', () => {
-    test.todo('errors on malformed input');
-    test.todo(
-      'creates a new epoch on the correct day of the correct week of the month'
-    );
-    test.todo('handles cases at the end of the month correctly');
-    test.todo(
-      'always extends the epoch to end on the start date of the following month'
-    );
+    describe('findSameDayNextMonth', () => {
+      it('handles month beginnings correctly', () => {
+        let date = DateTime.fromISO('2023-01-01T20:28:00.000Z');
+        let result = findSameDayNextMonth(date, {
+          week: 0,
+        });
+        expect(result.toISO()).toBe('2023-02-05T20:28:00.000Z');
+
+        date = DateTime.fromISO('2023-02-06T20:28:00.000Z');
+        result = findSameDayNextMonth(date, {
+          week: 0,
+        });
+        expect(result.toISO()).toBe('2023-03-06T20:28:00.000Z');
+
+        date = DateTime.fromISO('2023-03-06T20:28:00.000Z');
+        result = findSameDayNextMonth(date, {
+          week: 0,
+        });
+        expect(result.toISO()).toBe('2023-04-03T20:28:00.000Z');
+      });
+      it('handles month endings correctly', () => {
+        let date = DateTime.fromISO('2023-01-31T20:28:00.000Z');
+
+        let result = findSameDayNextMonth(date, {
+          week: 4,
+        });
+        expect(result.toISO()).toBe('2023-03-07T20:28:00.000Z');
+
+        date = DateTime.fromISO('2023-02-07T20:28:00.000Z');
+        result = findSameDayNextMonth(date, {
+          week: 5,
```

Comment:
I'm confused about the week arg i guess, If its the 5th week, why is the test from 02-07 and why do we need the week? Shouldn't the week of the month be calculated from the start date rather than provided?